gateway/src/http_entrypoints.rs

```diff
@@ -397,6 +397,15 @@ pub struct ImageVersion {
     pub version: u32,
 }
 
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub struct InstallinatorImageId {
+    pub host_os: [u8; 32],
+    pub control_plane: [u8; 32],
+}
+
```

Comment:
Hm, I'm not sure they should share. This one is part of MGS's HTTP API, and ipcc-key-value is for the SP/installinator path; they need to contain the same data but don't necessarily have to have the same format. Case in point - should I change these to `String` and have MGS unhexlify them? We definitely don't want hex strings in the CBOR as you pointed out above, but doing that transformation in MGS seems reasonable.

Reply:
OK, yeah, good point! Maybe add a comment saying that this is related to that then?

Reply:
Yeah, will do. There's also a `From` impl (which would need to become a `TryFrom` if we switch to hex). I'm on the fence on making this a `String` since it makes the client slightly less type safe, but it is weird to see hash passed as 32 ints in an API for sure. Do you still lean toward String/hex even if that means the client just takes `String` and MGS does the unhexlification "by hand"? (I'm assuming there isn't a _lot_ we can do on the progenitor side of this without a hexstring type in openapi.)

Reply:
I've gone back and forth on this for different things before also. It's frustrating to not be able to cut and paste a hexstring. I also think hexstrings are less precise/type safe. For now I'd stick with with the array type. If it becomes a problem we can probably write a decoder tool easily enough.